agent/stats/engine_unix_test.go

```diff
@@ -124,3 +130,79 @@ func TestNetworkModeStatsAWSVPCMode(t *testing.T) {
 		}
 	}
 }
+
+func TestGetContainerStatusMessage(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	testContainer := &apicontainer.DockerContainer{
+		DockerID: "123",
+		Container: &apicontainer.Container{
+			Name: "firelens_v2_container",
+			FirelensConfig: &apicontainer.FirelensConfig{
+				Version: "v2",
+			},
+		},
+	}
+
+	engine := NewDockerStatsEngine(&cfg, nil, nil)
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+	engine.ctx = ctx
+	engine.config.DataDirOnHost = os.TempDir()
+	engine.dockerIdToStatusMessageSince = map[string]time.Time{}
+
+	testCases := []struct {
+		name          string
+		statusMessage string
+		isNil         bool
+		modifyFile    bool
+		fileNotExists bool
+	}{
+		{
+			name:          "getContainerStatusMessage returns new message",
+			statusMessage: "new status message",
+			isNil:         false,
+			modifyFile:    true,
+		},
+		{
+			name:          "getContainerStatusMessage returns updated message",
+			statusMessage: "updated status message",
+			isNil:         false,
+			modifyFile:    true,
+		},
+		{
+			name:          "getContainerStatusMessage returns empty string for no update",
+			statusMessage: "",
+			isNil:         true,
+		},
+		{
+			name:          "getContainerStatusMessage returns empty string when status message file does not exist",
+			statusMessage: "",
+			isNil:         false,
+			fileNotExists: true,
+		},
+	}
+
+	tempDir := "/tmp/data/telemetry/abc/status-message/firelens_v2_container"
+	os.MkdirAll(tempDir, os.ModePerm)
+	tempfile, _ := os.Create(tempDir + "/status-message")
+	defer func() {
```

Comment:
nit: this can just be `defer os.Remove(tempfile.Name())`

Reply:
oops right! Fixed in the latest rev